### PR TITLE
Enforce password in public share dialog depending on new enforced_for

### PR DESCRIPTION
**Pull request by davigonz**
Implements https://github.com/owncloud/android/issues/2228
Needs https://github.com/owncloud/android-library/pull/237

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java [hunk 1]
```diff
@@ -745,11 +746,51 @@ private void updateInputFormAccordingToServerCapabilities() {
...
+                mReadWriteButton.isChecked() &&
+                        mCapabilities.getFilesSharingPublicPasswordEnforcedReadWrite().isTrue() ||
+                mUploadOnlyButton.isChecked() &&
+                        mCapabilities.getFilesSharingPublicPasswordEnforcedUploadOnly().isTrue()) {
+            setPasswordEnforced();
+        }
+
+        // Set password label depending on the checked permission option
+        mPermissionRadioGroup.setOnCheckedChangeListener(
+                (group, checkedId) -> {
```
Comment (hannesa2) [inline owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: Do you like this code style ?
If I would do it, I would join L760 and L761

File: owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java [hunk 2]
```diff
@@ -745,11 +746,51 @@ private void updateInputFormAccordingToServerCapabilities() {
                             mCapabilities.getFilesSharingPublicExpireDateDays()));
         }
 
+        // Set password label when opening the dialog
+        if (mReadOnlyButton.isChecked() &&
+                mCapabilities.getFilesSharingPublicPasswordEnforcedReadOnly().isTrue() ||
```
Comment (hannesa2) [inline owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: What is this ? `.isTrue()` Hard stuff !

File: owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java [hunk 3]
```diff
@@ -745,11 +746,51 @@ private void updateInputFormAccordingToServerCapabilities() {
...
+                        mCapabilities.getFilesSharingPublicPasswordEnforcedReadWrite().isTrue() ||
+                        mCapabilities.getFilesSharingPublicPasswordEnforcedUploadOnly().isTrue();
+
         // hide password switch if password is enforced to prevent it is removed
-        if (mCapabilities.getFilesSharingPublicPasswordEnforced().isTrue()) {
-            mPasswordLabel.setText(R.string.share_via_link_password_enforced_label);
-            mPasswordSwitch.setVisibility(View.GONE);
-            mPasswordValueEdit.setVisibility(View.VISIBLE);
+        if (!hasPasswordEnforcedFor &&
```
Comment (hannesa2) [inline owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: join to one line ?

File: owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java (hunk 1 above)
Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: Done

File: owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java (hunk 2 above)
Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: isTrue() checks if the capability value is TRUE(1)

File: owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java (hunk 3 above)
Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: done

**Comment by davigonz** [comment]
Thanks a lot @hannesa2 for the code review, just applied the changes you requested

**Comment by davigonz** [comment]
@jesmrec ready to test

**Comment by jesmrec** [comment]
Test performed: https://github.com/owncloud/QA/blob/master/Mobile/Android/Release_2.10.0/2445-enforce_password_public_links.md

Approved.

**Pull request merged**